diff -ru a/libraries/base/GHC/Event/EPoll.hsc b/libraries/base/GHC/Event/EPoll.hsc
--- a/libraries/base/GHC/Event/EPoll.hsc	2018-06-12 20:33:31.000000000 +0000
+++ b/libraries/base/GHC/Event/EPoll.hsc	2020-04-03 21:45:39.710810931 +0000
@@ -173,7 +173,7 @@
  , epollOut = EPOLLOUT
  , epollErr = EPOLLERR
  , epollHup = EPOLLHUP
- , epollOneShot = EPOLLONESHOT
+ , epollOneShot = 0x40000000  // EPOLLONESHOT
  }
 
 -- | Create a new epoll context, returning a file descriptor associated with the context.
diff -ru a/llvm-targets b/llvm-targets
--- a/llvm-targets	2019-04-07 19:39:58.000000000 +0000
+++ b/llvm-targets	2020-04-03 21:46:46.238673727 +0000
@@ -13,6 +13,7 @@
 ,("i386-unknown-linux", ("e-m:e-p:32:32-f64:32:64-f80:32-n8:16:32-S128", "pentium4", ""))
 ,("x86_64-unknown-linux-gnu", ("e-m:e-i64:64-f80:128-n8:16:32:64-S128", "x86-64", ""))
 ,("x86_64-unknown-linux", ("e-m:e-i64:64-f80:128-n8:16:32:64-S128", "x86-64", ""))
+,("arm-unknown-linux-androideabi", ("e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64", "generic", ""))
 ,("armv7-unknown-linux-androideabi", ("e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64", "generic", ""))
 ,("aarch64-unknown-linux-android", ("e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128", "generic", "+neon"))
 ,("powerpc64le-unknown-linux", ("e-m:e-i64:64-n32:64", "ppc64le", ""))
diff -ru a/mk/warnings.mk b/mk/warnings.mk
--- a/mk/warnings.mk	2019-04-07 19:39:58.000000000 +0000
+++ b/mk/warnings.mk	2020-04-03 21:47:13.878616492 +0000
@@ -1,5 +1,5 @@
 # See Note [Order of warning flags].
-SRC_CC_OPTS     += -Wall $(WERROR)
+SRC_CC_OPTS     += -std=gnu99 -Wall $(WERROR)
 SRC_HC_OPTS     += -Wall
 # Don't add -Werror to SRC_HC_OPTS_STAGE0 (or SRC_HC_OPTS), because otherwise
 # validate may unnecessarily fail when booting with an older compiler.
diff -ru a/utils/llvm-targets/gen-data-layout.sh b/utils/llvm-targets/gen-data-layout.sh
--- a/utils/llvm-targets/gen-data-layout.sh	2019-04-07 19:39:58.000000000 +0000
+++ b/utils/llvm-targets/gen-data-layout.sh	2020-04-03 21:46:32.610701898 +0000
@@ -28,7 +28,7 @@
     # Linux x86
     "i386-unknown-linux-gnu" "i386-unknown-linux" "x86_64-unknown-linux-gnu" "x86_64-unknown-linux"
     # Linux Android
-    "armv7-unknown-linux-androideabi" "aarch64-unknown-linux-android"
+    "arm-unknown-linux-androideabi" "armv7-unknown-linux-androideabi" "aarch64-unknown-linux-android"
     # Linux ppc64le
     "powerpc64le-unknown-linux"
 
