diff -ru a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2016-06-30 14:03:47.106158181 +0200
+++ b/src/Makefile.am	2016-06-30 14:04:13.982044861 +0200
@@ -51,30 +51,30 @@
 LIBOPTS = $(LIBOPTS_LDADD)
 endif
 
-bin_PROGRAMS = psktool gnutls-cli-debug certtool
+#bin_PROGRAMS = psktool gnutls-cli-debug certtool
 if ENABLE_SRP
-bin_PROGRAMS += srptool
+#bin_PROGRAMS += srptool
 endif
 
 if ENABLE_OCSP
-bin_PROGRAMS += ocsptool gnutls-serv 
+#bin_PROGRAMS += ocsptool gnutls-serv 
 if ENABLE_ANON
-bin_PROGRAMS += gnutls-cli
+#bin_PROGRAMS += gnutls-cli
 endif
 endif
 
 if ENABLE_DANE
-bin_PROGRAMS += danetool
+#bin_PROGRAMS += danetool
 endif
 
 if ENABLE_TROUSERS
-bin_PROGRAMS += tpmtool
+#bin_PROGRAMS += tpmtool
 endif
 
-noinst_PROGRAMS = systemkey
+#noinst_PROGRAMS = systemkey
 
 if ENABLE_PKCS11
-bin_PROGRAMS += p11tool
+#bin_PROGRAMS += p11tool
 endif
 
 noinst_LTLIBRARIES =
